multimodal/src/autogluon/multimodal/optimization/lit_distiller.py

```diff
@@ -220,18 +220,20 @@ def _compute_loss(
 
         return loss
 
-    def _compute_metric(
+    def _compute_metric_score(
         self,
+        metric: torchmetrics.Metric,
+        custom_metric_func: Callable,
         logits: torch.Tensor,
         label: torch.Tensor,
     ):
-        if isinstance(self.validation_metric, torchmetrics.AUROC):
+        if isinstance(metric, (torchmetrics.AUROC, torchmetrics.AveragePrecision)):
             prob = F.softmax(logits.float(), dim=1)
-            return self.validation_metric(preds=prob[:, 1], target=label)  # only for binary classification
-        elif isinstance(self.validation_metric, BaseAggregator):
-            return self.validation_metric(self.custom_metric_func(logits, label))
+            metric.update(preds=prob[:, 1], target=label)  # only for binary classification
```

Comment:
Should we check the "full_state_update" flag in the metrics and decide how to trigger update?

See also https://torchmetrics.readthedocs.io/en/stable/pages/implement.html#internal-implementation-details .

Reply:
`full_state_update ` is used in `.forward()`. Since we only need `.update()`, we don't need it.